### PR TITLE
[mbr] Add Apple sample

### PR DESCRIPTION
**Pull request by lambdageek**
Hot reload for iOS Simulator and Mac Catalyst

Contributes to #44806

### CODE REVIEW

File: src/mono/sample/mbr/DeltaHelper/DeltaHelper.cs [hunk 1]
```diff
@@ -48,7 +48,8 @@ public class DeltaHelper {
 
         private static void LoadMetadataUpdate (Assembly assm, byte[] dmeta_data, byte[] dil_data, byte[] dpdb_data)
         {
-            UpdateMethod (assm, dmeta_data, dil_data, dpdb_data);
+            // UpdateMethod (assm, dmeta_data, dil_data, dpdb_data);
+	    System.Reflection.Metadata.AssemblyExtensions.ApplyUpdate (assm, dmeta_data, dil_data, dpdb_data);
```
Comment (CoffeeFlux) [inline src/mono/sample/mbr/DeltaHelper/DeltaHelper.cs:52]: Indentation looks wrong here, are you mixing tabs and spaces?

**Review by CoffeeFlux — COMMENTED** [review]
At a glance, you appear to have indentation issues in most of these files. I pointed out one such example.

File: src/mono/sample/mbr/DeltaHelper/DeltaHelper.cs (hunk 1 above)
Comment (lambdageek) [reply src/mono/sample/mbr/DeltaHelper/DeltaHelper.cs:52]: bah. yes. and also I meant to delete a bunch of the old code in here.

**Review by CoffeeFlux — APPROVED** [review]
Didn't test locally, but generally LGTM.

**Pull request merged**